bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -1199,3 +1210,40 @@ impl TryFrom<ImageInfo> for RumaAvatarImageInfo {
         }))
     }
 }
+
+struct PollData {
+    question: String,
+    answers: Vec<String>,
+    max_selections: u8,
+    poll_kind: PollKind,
+}
+
+impl PollData {
+    fn fallback_text(&self) -> String {
+        self.answers.iter().enumerate().fold(self.question.clone(), |acc, (index, answer)| {
+            format!("{acc}\n{}. {answer}", index + 1)
+        })
+    }
```

Comment:
Preexisting, but we could avoid most implicit allocations (from `format!`) by using `write!` here, which appends to an existing string (hat tip to Jonas who suggested it in internal channel):

```suggestion
        self.answers.iter().enumerate().fold(self.question.clone(), |acc, (index, answer)| {
            write!(&mut acc, "\n{}. {answer}", index + 1).unwrap();
            acc
        })
    }
```

(This requires the `std::fmt::Write` trait to be in scope.)

Reply:
Applied [here](https://github.com/matrix-org/matrix-rust-sdk/pull/2854/commits/b97f58124aa9d180296cd606777b898a4044661c)